### ISSUE TITLE
can we build hierarchy?

### ISSUE THREAD

**Issue by vetik32** [AlexKhymenko/ngx-permissions#34]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[x] Feature request
[x] Documentation issue or request
</code></pre>

## Expected behavior

it's possible to build hierarchical rules? 

```html
you can: 
<div *ngxPermissionsOnly="['CREATE']">
    <span class="btn">create new</span>
</div>
<div *ngxPermissionsOnly="['READ']">
    <span class="btn">view</span>
</div>
<div *ngxPermissionsOnly="['UPDATE']">
    <span class="btn">edit</span>
</div>
<div *ngxPermissionsOnly="['DELETE']">
    <span class="btn">delete</span>
</div>
```
which should:
 show all 4 buttons in case you have DELETE role/permission - `CRUD`
 show 3 buttons in case you have UPDATE role/permission `CRU`
 show 2 buttons in case you have Create/Read role/permission `CR`
 show 1 button in case you have Read role/permission `R`

```js
    this.persmissionService.addPermission('READ');
    this.rolesService.addRole('CREATE', ['READ'])
    this.rolesService.addRole('UPDATE', ['CREATE'])
    this.rolesService.addRole('DELETE', ['UPDATE'])
```
in other words considering this hierarchy `delete > update > create > read` 
will the *div* below show up if user has "DELETE" permission?

```html
<div *ngxPermissionsOnly="['READ']">
    <span class="btn">view</span>
</div>
```
or it can be achieved by repeating 
```js
    this.rolesService.addRole('DELETE', ['CREATE', 'READ', 'UPDATE'])
    this.rolesService.addRole('UPDATE', ['CREATE', 'READ'])
    this.rolesService.addRole('CREATE', ['READ'])

```

**Comment by AlexKhymenko** [comment]
Use 2 approach for sure its easier to read and to maintain :-)
